ejb/WFLY-17986.adoc

```diff
@@ -0,0 +1,234 @@
+contained within an EJB-based application deployment on EAP: the client obtains a proxy for the specific bean
+in the deployment it wants to interact with, and uses the proxy to make invocations on the bean.
+However, over and above this basic invocation mechanism, the EJB client provides a number of important features,
+depending on the type of deployment.
+
+The key features for both HA and non-HA deployments include:
+
+* lookup via JNDI or programmatic creation of proxies for EJBs in a deployment
+* client-specific security and discovery configuration
+* transactional scoping of invocations on EJBs in a deployment
+
+Additional features for HA deployments:
+
+* load balancing - requests to SLSBs can be load balanced across the Pods in the EAP deployment's cluster
+* affinity management - requests to SFSBs will be routed to the specific Pod in the EAP cluster on which the session
+was initially created; in other words, the client proxy for the SFSB will have an affinity to a particular backend Pod
+and that affinity will be maintained
+* failover - requests to SFSBs will be retried on a different Pod in the EAP cluster if the Pod processing a request
+fails, and the affinity will be updated
+
+A major part of this issue is to ensure that these key features work in OpenShift as they do in bare metal.
+
+=== The WildFly operator
+The OpenShift environment differs from a bare metal environment in many key ways; for example, in OpenShift, Pod IP
+addresses and storage volume references are transient, whereas on bare metal they are persistent; in OpenShift,
+Pods which fail can be restarted automatically, whereas on bare metal, they are under the control of a system
+administrator; and in OpenShift, the number of instances of Pods backing a service can be scaled up or scaled down
+automatically. In other words, the OpenShift environment is more dynamic than a corresponding bare metal environment.
+
+Unless otherwise constrained, it would be very difficult for an EJB client application to interact with an EAP
+deployment in OpenShift if stricter guarantees were not imposed. For example, discovery of the members of a
+EAP cluster and affinity management of SFSBs would be next to impossible without an assumption persistent Pod
+IP addresses.
+
+The `WildFly operator` can be used to deploy an EAP application in OpenShift and provide the following guarantees:
+
+* persistent IP addresses for Pods
+* persistent storage references for storage used by EAP
+* scaling up or scaling down of services which are transaction-aware
+
+In what follows, we assume that deployments are made using the WildFly operator and that the EJB client may depend
+on these additional guarantees. For more information on the WildFly operator, see
+https://operatorhub.io/operator/WildFly[WildFly Operator on operatorhub.io]
+
+=== Scope
+As mentioned earlier, the scope of this issue is limited to `external` EJB clients using the `http` protocol to
```

Comment:
@rachmatowicz where are we going to tests HELM based deployments? in some different RFE?